src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,325 @@
+
+            SendCommand(new SetSegmentReMapColorDepth());
+        }
+
+        /// <summary>
+        /// Initializes new instance of Ssd1351 device that will communicate using SPI bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="dataCommandPin">The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPin">The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        /// <param name="spiBufferSize">The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions. The default value is 4096.</param>
+        public Ssd1351(SpiDevice spiDevice, int dataCommandPin, int resetPin, int spiBufferSize = DefaultSPIBufferSize) : this(spiDevice, new GpioController(), dataCommandPin, resetPin, spiBufferSize)
+        {
+            //as a default GPIO controller has been used then dispose of it later..
+            _disposeGpioController = true;
+        }
+
+        /// <summary>
+        /// Convert a color structure to a byte tuple represening the colour in 565 format.
+        /// </summary>
+        /// <param name="color">The color to be converted.</param>
+        private (byte, byte) Color565(Color color)
+        {
+            // get the top 5 MSB of the blue or red value
+            UInt16 retval = (UInt16)((_colorSequence == ColorSequence.ABC ? color.B : color.R) >> 3);
+            // shift right to make room for the green Value
+            retval <<= 6;
+            // combine with the 6 MSB if the green value
+            retval |= (UInt16)(color.G >> 2);
+            // shift right to make room for the red or blue Value
+            retval <<= 5;
+            // combine with the 6 MSB if the red or blue value
+            retval |= (UInt16)((_colorSequence == ColorSequence.ABC ? color.R : color.B) >> 3);
+            return ((byte)(retval >> 8), (byte)(retval & 0xFF));
+        }
+
+        /// <summary>
+        /// Send filled rectangle to the ssd1351 display.
+        /// </summary>
+        /// <param name="color">The color to fill the rectangle with.</param>
+        /// <param name="x">The x co-ordinate of the point to start the rectangle at.</param>
+        /// <param name="y">The y co-ordinate of the point to start the rectangle at.</param>
+        /// <param name="w">The width of the rectangle.</param>
+        /// <param name="h">The height of the rectangle.</param>
+        public void FillRect(Color color, byte x, byte y, byte w, byte h)
+        {
+
+            byte[] colourBytes = new byte[_colorDepth == ColorDepth.ColourDepth65K ? 2 : 3]; // create a short array that holds the colour data to be sent to the display
```

Comment:
Instead of all of the allocations for byte arrays that we are doing here, we should instead just use Span<byte> in order to not stress out the GC if it is not necessary. For that I mean
```c#
Span<byte> colourBytes = stackalloc byte[_colorDepth == ColorDepth.ColourDepth65K ? 2 : 3];
```

Reply:
@joperezr 
Done in the latest PR update. Also had a look at most of the code for issues like this